liquibase-core/src/main/java/liquibase/structure/DatabaseObjectCollection.java

```diff
@@ -75,7 +74,7 @@ public void add(DatabaseObject databaseObject) {
         }
         Map<String, Set<DatabaseObject>> collectionMap = cache.get(databaseObject.getClass());
         if (collectionMap == null) {
-            collectionMap = new HashMap<String, Set<DatabaseObject>>();
+            collectionMap = new LinkedHashMap<>();
```

Comment:
If I understand correctly key of this LinkedHashMap is object name, so order for these objects is going to be preserved by object type, why do we need this change? i.e. I can't fully understand how it helps to preserve specific objects ordering, i.e. that some_func_1 is going before some_func_2